src/components/ConnectModal/ConnectWithWalletConnect/index.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useCallback, useEffect } from "react";
+import { PageContent } from "../../Common/Modal/styles";
+import { useContext } from "../../AlephiumConnect";
+import { Container } from "../ConnectWithInjector/styles";
+import {
+  WalletConnectProvider,
+  QRCodeModal,
+} from "@alephium/walletconnect-provider";
+
+let _init = false;
+
+const ConnectWithWalletConnect: React.FC = () => {
+  const context = useContext();
+
+  const onQrClose = useCallback(() => {
+    console.log("qr closed");
+  }, []);
+
+  const wcConnect = async () => {
+    const wcProvider = await WalletConnectProvider.init({
+      projectId: "6e2562e43678dd68a9070a62b6d52207",
+      networkId: 0,
+    });
+
+    wcProvider.on("displayUri", (uri) => {
+      context.setOpen(false);
+      QRCodeModal.open(uri, onQrClose);
+    });
+
+    try {
+      await wcProvider.connect();
```

Comment:
Do you think it makes sense to handle connect in `useConnect` hook? From the users perspective we either pick extension wallet or wallet connect from the popup window and from that point on `useConnect` and `useAccount` hook remembers the decision and will use the right connector. It also auto-reconnects after refreshing the page.

Reply:
The state of those hook will be lost once the page is refreshed right? If that's the case, then we may need to introduce storage.